diff --git a/ports/rp2/CMakeLists.txt b/ports/rp2/CMakeLists.txt
index 4334a0a..0b010fe 100644
--- a/ports/rp2/CMakeLists.txt
+++ b/ports/rp2/CMakeLists.txt
@@ -411,6 +411,15 @@ target_compile_options(${MICROPY_TARGET} PRIVATE
 target_link_options(${MICROPY_TARGET} PRIVATE
     -Wl,--defsym=__micropy_c_heap_size__=${MICROPY_C_HEAP_SIZE}
 )
+# Do not include stack unwinding & exception handling for C++ user modules
+target_compile_definitions(usermod INTERFACE PICO_CXX_ENABLE_EXCEPTIONS=0 PICO_MAX_SHARED_IRQ_HANDLERS=8)
+target_compile_options(usermod INTERFACE $<$<COMPILE_LANGUAGE:CXX>:
+    -fno-exceptions
+    -fno-unwind-tables
+    -fno-rtti
+    -fno-use-cxa-atexit
+>)
+target_link_options(usermod INTERFACE -specs=nano.specs)
 
 set_source_files_properties(
     ${PICO_SDK_PATH}/src/rp2_common/pico_double/double_math.c
